test/integration/complex.js

```diff
@@ -0,0 +1,88 @@
+var fs = require('fs');
+var path = require('path');
+var test = require('tap').test;
+var extract = require('../fixtures/extract');
+var VirtualMachine = require('../../src/index');
+
+var projectUri = path.resolve(__dirname, '../fixtures/complex.sb2');
+var project = extract(projectUri);
+
+var spriteUri = path.resolve(__dirname, '../fixtures/sprite.json');
+var sprite = fs.readFileSync(spriteUri, 'utf8');
+
+test('complex', function (t) {
+    var vm = new VirtualMachine();
+
+    // Evaluate playground data and exit
+    vm.on('playgroundData', function (e) {
+        var threads = JSON.parse(e.threads);
+        t.ok(threads.length === 0);
+        t.end();
+        process.nextTick(process.exit);
+    });
+
+    // Manipulate each target
+    vm.on('targetsUpdate', function (data) {
+        var targets = data.targetList;
+        for (var i in targets) {
+            if (targets[i].isStage === true) continue;
+            if (targets[i].name === 'test') continue;
```

Comment:
Note for the future: at some point `vm.renameSprite` will probably disambiguate two sprites with the same name, at which point this might need to change to check if `targets[i].name` starts with `'test'` or something like that.

That said... it [looks like](https://github.com/LLK/scratch-vm/blob/dc9126e0676d8b6a565982f1c0616203c3e2a447/src/index.js#L308) `targetsUpdate` gives you a frozen JSON copy of the targets, so the rename below shouldn't cause this `continue` to be executed even if the targets move around in `runtime.targets`. Is that what this check was added for, or something else?